sys/posix/pthread/Kconfig

```
@@ -8,3 +8,11 @@
 config MODULE_PTHREAD
     bool "Pthread Support"
     depends on TEST_KCONFIG
+    select MODULE_PERIPH_TIMER
+    select MODULE_TIMEX
+    select MODULE_ZTIMER
+    select MODULE_ZTIMER_PERIPH_TIMER
+    select MODULE_ZTIMER64
+    select MODULE_ZTIMER64_INIT
+    select MODULE_ZTIMER64_USEC
+    select MODULE_ZTIMER64_XTIMER_COMPAT if MODULE_ZTIMER_XTIMER_COMPAT
```

```suggestion
    select MODULE_ZTIMER
    select ZTIMER64_USEC
    select MODULE_ZTIMER64_XTIMER_COMPAT if MODULE_ZTIMER_XTIMER_COMPAT
    select MODULE_TIMEX
```

A subtle change can make the difference.

---

I think I tried your proposal and there were still missing modules compared to make => that's the reason of my extended changes.

---

Due to circular dependancies in the timer abstraction layer we have a non module symbol that brings in many module symbols. This is the `ZTIMER64_USEC` vs the `MODULE_ZTIMER64_USEC`.

My guess is you used module version instead of applying the diff.

---

> My guess is you used module version instead of applying the diff.

Hmm ok, that looked like a typo... but is in fact yet another hack

---

Originally I was thinking to make it more explicit with a `CIRCULAR_MODULE_*` or something.

This was a tough one, again, we welcome input for a way around it.  Ideally we would not have the circular dependency at all.

---

Ok, finally applied your initial proposal.